Sources/Commands/PackageCommands/AuditBinaryArtifact.swift

```diff
@@ -0,0 +1,136 @@
+        for binary in try await detectDefaultObjects(
+            clang: clang, fileSystem: fileSystem, hostTriple: hostTriple)
+        {
+            try await symbolProvider.symbols(
+                for: binary, symbols: &hostDefaultSymbols, recordUndefined: false)
+        }
+
+        let extractedArtifact = try await extractArtifact(
+            fileSystem: fileSystem, scratchDirectory: swiftCommandState.scratchDirectory)
+
+        guard
+            let artifactKind = try Workspace.BinaryArtifactsManager.deriveBinaryArtifactKind(
+                fileSystem: fileSystem,
+                path: extractedArtifact,
+                observabilityScope: swiftCommandState.observabilityScope
+            )
+        else {
+            throw StringError("Invalid binary artifact provided at \(path)")
+        }
+
+        let module = BinaryModule(
+            name: path.basenameWithoutExt, kind: artifactKind, path: extractedArtifact,
+            origin: .local)
+        for library in try module.parseLibraryArtifactArchives(
+            for: hostTriple, fileSystem: fileSystem)
+        {
+            var symbols = hostDefaultSymbols
+            try await symbolProvider.symbols(for: library.libraryPath, symbols: &symbols)
+
+            guard symbols.undefined.isEmpty else {
+                print(
+                    "Invalid artifact binary \(library.libraryPath.pathString), found undefined symbols:"
+                )
+                for name in symbols.undefined {
+                    print("- \(name)")
+                }
+                throw ExitCode(1)
+            }
+        }
+
+        print(
+            "Artifact is safe to use on the platforms runtime compatible with triple: \(hostTriple.tripleString)"
+        )
+    }
+
+    private func extractArtifact(fileSystem: any FileSystem, scratchDirectory: AbsolutePath)
+        async throws -> AbsolutePath
+    {
+        let archiver = UniversalArchiver(fileSystem)
+
+        guard let lastPathComponent = path.components.last,
+            archiver.isFileSupported(lastPathComponent)
+        else {
+            let supportedExtensionList = archiver.supportedExtensions.joined(separator: ", ")
+            throw StringError(
+                "unexpected file type; supported extensions are: \(supportedExtensionList)")
+        }
+
+        // Ensure that the path with the accepted extension is a file.
+        guard fileSystem.isFile(path) else {
+            throw StringError("file not found at path: \(path.pathString)")
+        }
+
+        let archiveDirectory = scratchDirectory.appending(
+            components: "artifact-auditing",
+            path.basenameWithoutExt, UUID().uuidString
+        )
+        try fileSystem.forceCreateDirectory(at: archiveDirectory)
+
+        try await archiver.extract(from: path, to: archiveDirectory)
+
+        let artifacts = try fileSystem.getDirectoryContents(archiveDirectory)
```

Comment:
Just to check are we filtering somewhere to only use the `.a` that also applies to the current host triplet?

Reply:
`parseLibraryArtifactsArchive` does it for us.